rollup.config.js

```diff
@@ -2,60 +2,79 @@
 import babel from '@rollup/plugin-babel';
 // Used to minify files
 import { terser } from "rollup-plugin-terser";
+import typescript from '@rollup/plugin-typescript';
+
 import pkg from './package.json';
 
-export default [{
-  input: 'src/index.js',
-  output: { // ES2015 (ES6) Module
-    file: pkg.module,
-    format: 'es',
-    sourcemap: true,
-  },
-  plugins: [terser()]
-},{
-  // Outputs we want to be transpiled by Babel
-  input: 'src/index.js',
-  output: [
-    { // Common JS
-      file: pkg.main,
-      format: 'cjs',
+export default [
+  {
+    input: 'src/index.ts',
+    output: { // ES2015 (ES6) Module
+      file: pkg.module,
+      format: 'es',
       sourcemap: true,
     },
-    { // Browser
-      name: 'ExportedTests',
-      file: pkg.browser,
-      format: 'umd',
-      sourcemap: true,
-    }
-  ],
-  plugins: [
-    babel({ babelHelpers: 'bundled' }),
-    terser()
-  ]
-},
-
-/**
- * DEPRECIATED FILES
- * @todo remove these on the next major release
- */
-{
-  input: 'src/index.js',
-  output: { // ES2015 (ES6) Module
-    file: 'index.js',
-    format: 'es',
+    plugins: [
+      // Can't figure out how to get Rollup to create the declaration files (.d.ts)
+      // with the TypeScript plugin. Trying to define the declaration configs in the
+      // tsconfig.json file creates errors for Rollup; defining the configurations in
+      // Rollup has required properties that conflict with Rollup. I'm not sure if
+      // having them created separately by TypeScript will allow our users to properly
+      // map to the minified files created by Rollup.
+      typescript(),
```

Comment:
AFAIK rollup doesn't have any provisions for generating `d.ts` ... but it really doesn't need to, since `tsc` can handle the primary compilation ... as long as the types are accessible (usually pointed to with the `types` key in the `package.json`, or side by side with the `.js` files IIRC) users shouldn't have any issues resolving the definitions.